README.md

```diff
@@ -164,27 +164,44 @@ reading the following articles:
 - [Functional Swift for Dealing with Optional Values](http://robots.thoughtbot.com/functional-swift-for-dealing-with-optional-values)
 - [Railway Oriented Programming](http://fsharpforfunandprofit.com/posts/recipe-part2/)
 
+And check out this talk:
+
+- [How I Learned To Stop Worrying And Love The Functor](https://github.com/gfontenot/talks/tree/master/Functors)
+
 ## Usage
 
-The first thing you need to do when you receive JSON data is convert it to an
-instance of the `JSONValue` enum. This is done by passing the `AnyObject`
-value returned from `NSJSONSerialization` to `JSONValue.parse()`:
+The first thing you need to do when you receive JSON data is convert it from
+`NSData` to an `AnyObject` using the build in `NSJSONSerialization` API. Once
+you have the `AnyObject`, you can call the global `decode` function to get back
+the decoded model.
 
 ```swift
-let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(responseData, options: NSJSONReadingOptions(0), error: nil)
+var error: NSError?
+let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(responseData, options: NSJSONReadingOptions(0), error: &error)
 
 if let j: AnyObject = json {
-  let value: JSONValue = JSONValue.parse(j)
-  let user: User? = User.decode(value)
+  let user: User? = decode(j) // ignore error info or
+  let decodedUser: Decoded<User> = decode(j) // preserve error info
+} else {
+  // handle error
 }
 ```
 
 Note that you probably want to use an error pointer to track errors from
 `NSJSONSerialization`.
 
-The `JSONValue` enum exists to help with some of the type inference, and also
-wraps up some of the casting that we'll need to do to transform the JSON into
-native types.
+The `JSON` enum exists to help with some of the type inference, and also wraps
+up some of the casting that you'll need to do to transform the JSON into native
+types.
+
+Argo 1.0 introduces a new type: `Decoded<T>`. This is now the type returned
+from the `decode` function that you implement as part of the `JSONDecodable`
```

Comment:
All these instances of `JSONDecodable` have to be replaced with `Decodable` now, correct?